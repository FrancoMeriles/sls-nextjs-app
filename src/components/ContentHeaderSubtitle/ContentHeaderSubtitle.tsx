import { Box, Text } from '@chakra-ui/react'

const ContentHeaderSubtitle = ({ condition, sold_quantity }) => {
  return (
    <Box d="flex" alignItems="center">
      <Text fontSize="xs" color="gray.500">
        {condition === 'new' ? 'Nuevo' : 'Usado'}
      </Text>
      <Text fontSize="xs" color="gray.500" marginLeft={1}>
        | {sold_quantity && sold_quantity.toLocaleString('de')} Vendidos
      </Text>
    </Box>
  )
}

export default ContentHeaderSubtitle
