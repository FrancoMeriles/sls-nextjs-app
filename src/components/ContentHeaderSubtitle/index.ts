import ContentHeaderSubtitle from './ContentHeaderSubtitle'

export default ContentHeaderSubtitle
