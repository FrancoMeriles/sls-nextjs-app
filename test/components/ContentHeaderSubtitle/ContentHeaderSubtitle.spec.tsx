import { render, screen } from '@testing-library/react'
import ContentHeaderSubtitle from '@base/components/ContentHeaderSubtitle'
import renderer from 'react-test-renderer'

const mockContentHeaderSubtitleProps = {
  condition: 'new',
  sold_quantity: 500,
}

describe('ContentHeaderSubtitle', () => {
  it('should render and match snapshot with required props', () => {
    const tree = renderer
      .create(<ContentHeaderSubtitle {...mockContentHeaderSubtitleProps} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders and display the correct condition if is "new"', async () => {
    mockContentHeaderSubtitleProps.sold_quantity = 300

    render(<ContentHeaderSubtitle {...mockContentHeaderSubtitleProps} />)
    expect(screen.getByText('Nuevo')).toBeInTheDocument()
    expect(screen.getByText('| 300 Vendidos')).toBeInTheDocument()
  })

  it('should renders and display the correct condition if is "used"', async () => {
    mockContentHeaderSubtitleProps.condition = 'use'
    mockContentHeaderSubtitleProps.sold_quantity = 400

    render(<ContentHeaderSubtitle {...mockContentHeaderSubtitleProps} />)
    expect(screen.getByText('Usado')).toBeInTheDocument()
    expect(screen.getByText('| 400 Vendidos')).toBeInTheDocument()
  })
})
