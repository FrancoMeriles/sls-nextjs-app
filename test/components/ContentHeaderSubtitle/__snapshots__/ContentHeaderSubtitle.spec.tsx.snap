// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentHeaderSubtitle should render and match snapshot with required props 1`] = `
<div
  className="css-70qvj9"
>
  <p
    className="chakra-text css-1bmfkgj"
  >
    Nuevo
  </p>
  <p
    className="chakra-text css-kiw2af"
  >
    | 500 Vendidos
  </p>
</div>
`;
